### ISSUE TITLE
Mypy doesn't work with the latest release of lxml

### ISSUE THREAD

**Issue by rowillia** [JukkaL/mypy#1002]
Using mypy reports with the latest release of lxml results in the following stack trace:

```
$ mypy --html-report mypy_dir defaultdict_example.py
Traceback (most recent call last):
  File "/Users/rwilliams/src/oss/mypy/.venv/bin/mypy", line 6, in <module>
    main(__file__)
  File "/Users/rwilliams/src/oss/mypy/.venv/lib/python3.5/site-packages/mypy_lang-0.2.0.dev0-py3.5.egg/mypy/main.py", line 91, in main
    type_check_only(sources, bin_dir, options)
  File "/Users/rwilliams/src/oss/mypy/.venv/lib/python3.5/site-packages/mypy_lang-0.2.0.dev0-py3.5.egg/mypy/main.py", line 134, in type_check_only
    python_path=options.python_path)
  File "/Users/rwilliams/src/oss/mypy/.venv/lib/python3.5/site-packages/mypy_lang-0.2.0.dev0-py3.5.egg/mypy/build.py", line 197, in build
    result = manager.process(initial_states)
  File "/Users/rwilliams/src/oss/mypy/.venv/lib/python3.5/site-packages/mypy_lang-0.2.0.dev0-py3.5.egg/mypy/build.py", line 383, in process
    next.process()
  File "/Users/rwilliams/src/oss/mypy/.venv/lib/python3.5/site-packages/mypy_lang-0.2.0.dev0-py3.5.egg/mypy/build.py", line 857, in process
    self.manager.reports.file(self.tree, type_map=self.manager.type_checker.type_map)
  File "/Users/rwilliams/src/oss/mypy/.venv/lib/python3.5/site-packages/mypy_lang-0.2.0.dev0-py3.5.egg/mypy/report.py", line 41, in file
    reporter.on_file(tree, type_map)
  File "/Users/rwilliams/src/oss/mypy/.venv/lib/python3.5/site-packages/mypy_lang-0.2.0.dev0-py3.5.egg/mypy/report.py", line 138, in on_file
    xml_pi = etree.ProcessingInstruction('xml', 'version="1.0" encoding="utf-8"')
  File "src/lxml/lxml.etree.pyx", line 3039, in lxml.etree.ProcessingInstruction (src/lxml/lxml.etree.c:81073)
ValueError: Invalid PI name 'b'xml''
```

**Comment by JukkaL** [comment]
Thanks for reporting this! I'd like to get rid of the lxml dependency as soon as possible, but for now we need to figure how to live with it.

**Comment by WeeBull** [comment]
I've just hit the same thing. I think the correct thing here is to remove the `xml_pi = ProcessingInstruction...` line, and the line just after which uses `xml_pi`.

This appears to be a misunderstanding about the `<?xml version="1.0" encoding="UTF-8" ?>` line in XML files. It [is not a processing instruction](https://en.wikipedia.org/wiki/Processing_Instruction) despite often being called one. This is why it's explicitly checked for in recent lxml releases.

Reports then seem to work, or at least HTML ones do.

**Comment by gvanrossum** [comment]
Thanks @WeeBull! #1120 prevents the stated error with lxml version 3.5.0.
But the generated XML files don't contain <?xml version="1.0"
encoding="UTF-8" ? -- is that a problem?

On Tue, Jan 12, 2016 at 7:24 AM, WeeBull <notifications@github.com> wrote:

> I've just hit the same thing. I think the correct thing here is to remove
> the xml_pi = ProcessingInstruction... line, and the line just after which
> uses xml_pi.
>
> This appears to be a misunderstanding about the <?xml version="1.0"
> encoding="UTF-8" ?> line in XML files. It is not a processing instruction
> <https://en.wikipedia.org/wiki/Processing_Instruction> despite often
> being called one. This is why it's explicitly checked for in recent lxml
> releases.
>
> Reports then seem to work, or at least HTML ones do.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JukkaL/mypy/issues/1002#issuecomment-170945987>.
>



-- 
--Guido van Rossum (python.org/~guido)

**Comment by gvanrossum** [comment]
Fixed by 86974b199da6653e788710ea7762f319563f1ead.
